src/exp.h

```diff
@@ -69,7 +69,7 @@ inline const RegEx& Hex() {
 inline const RegEx& NotPrintable() {
   static const RegEx e =
       RegEx(0) |
-      RegEx("\x01\x02\x03\x04\x05\x06\x07\x08\x0B\x0C\x7F", REGEX_OR) |
+      RegEx("\x01\x02\x03\x04\x05\x06\x07\x08\x0B\x0C\x0D\x7F", REGEX_OR) |
```

Comment:
`0xD` is considered printable, according to the spec:

https://yaml.org/spec/1.2/spec.html#printable%20character//

Reply:
Yes, you are right. It is not reasonable to place `\x0D` in `NotPrintable()`. 
May I establish a  new `Exp::CR()` or  `Exp::CarriageReturn()`  to achieve the escape of `\r`, like the `Exp::Tab()`?

https://github.com/jbeder/yaml-cpp/blob/3f381f13a07f007941b250e1563f67ded4dc8b1a/src/emitterutils.cpp#L179

Reply:
I don't think this is correct; where in the spec does it disallow \r from appearing in a plain scalar?

Reply:
I have two reasons to do this:
1.  in the 5.1 chapter of spec, it says `Any excluded characters must be presented using escape sequences. In addition, any allowed characters known to be non-printable should also be escaped. `. I think `CR` belong to this.
2.  If the `\r` is allow to appear in a plain scallar, the "aaa\rbbb" will be parsed as "bbb". It may be not an expected result.